management/server/policy.go

```diff
@@ -243,175 +170,102 @@ type FirewallRule struct {
 
 	// Port of the traffic
 	Port string
-
-	// id for internal purposes
-	id string
 }
 
-// parseFromRegoResult parses the Rego result to a FirewallRule.
-func (f *FirewallRule) parseFromRegoResult(value interface{}) error {
-	object, ok := value.(map[string]interface{})
-	if !ok {
-		return fmt.Errorf("invalid Rego query eval result")
-	}
-
-	peerID, ok := object["ID"].(string)
-	if !ok {
-		return fmt.Errorf("invalid Rego query eval result peer ID type")
-	}
-
-	peerIP, ok := object["IP"].(string)
-	if !ok {
-		return fmt.Errorf("invalid Rego query eval result peer IP type")
-	}
-
-	direction, ok := object["Direction"].(string)
-	if !ok {
-		return fmt.Errorf("invalid Rego query eval result peer direction type")
-	}
-
-	action, ok := object["Action"].(string)
-	if !ok {
-		return fmt.Errorf("invalid Rego query eval result peer action type")
-	}
-
-	if v, ok := object["Protocol"]; ok {
-		if protocol, ok := v.(string); ok {
-			f.Protocol = protocol
-		}
-	}
-
-	if v, ok := object["Port"]; ok {
-		if port, ok := v.(string); ok {
-			f.Port = port
-		}
-	}
-
-	f.PeerID = peerID
-	f.PeerIP = peerIP
-	f.Direction = direction
-	f.Action = action
-
-	// NOTE: update this id each time when new field added
-	f.id = peerID + peerIP + direction + f.Protocol + f.Port + action
-
-	return nil
-}
-
-// queryPeersAndFwRulesByRego returns a list associated Peers and firewall rules list for this peer.
-func (a *Account) queryPeersAndFwRulesByRego(
-	peerID string,
-	queryNumber int,
-	query string,
-) ([]*Peer, []*FirewallRule) {
-	input := map[string]interface{}{
-		"peer_id": peerID,
-		"peers":   a.Peers,
-		"groups":  a.Groups,
-	}
-
-	stmt, err := rego.New(
-		rego.Query("data.netbird.all"),
-		rego.Module("netbird", defaultPolicyModule),
-		rego.Module(fmt.Sprintf("netbird-%d", queryNumber), query),
-	).PrepareForEval(context.TODO())
-	if err != nil {
-		log.WithError(err).Error("get Rego query")
-		return nil, nil
-	}
-
-	evalResult, err := stmt.Eval(
-		context.TODO(),
-		rego.EvalInput(input),
-	)
-	if err != nil {
-		log.WithError(err).Error("eval Rego query")
-		return nil, nil
-	}
+// getPeersByPolicy returns all peers that given peer has access to
```

Comment:
as we are getting peers and firewall rules, we can probably rename the method to `getPeerConnectionResources` and modify the comments. 

Another thing, the returning definition, `(peers []*Peer, rules []*FirewallRule)`, would be better: `(connPeers []*Peer, fwRules []*FirewallRule)`

Reply:
I renamed it, simplified this function, and extracted the closure function as a generator and another as a regular helper function. Add comments to all of them.